analyse_heterogeneous_sequences.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+from dark.fasta import FastaReads
+from math import log10
+from heterogeneous_sequences import heterogeneousSites, compareToRef
+import argparse
+import sys
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--homogenFraction", type=float, default=1.0,
+                    help="If the fraction of the most common nucleotide at a site is at least this value, the site will be considered homogenous.")
+parser.add_argument("--reference",
+                    help="Filename of the reference fasta sequence.")
+parser.add_argument("--reads", default=sys.stdin,
+                    help="Filename of the reference fasta sequence.")
+args = parser.parse_args()
+
+if 0 < args.homogenFraction < 1:
+    raise ValueError('--homogenFraction needs to be between 0 and 1.')
+
+reads = list(FastaReads(args.reads))
+
+if len(set(len(read) for read in reads)) != 1:
+    raise ValueError('Not all read lengths are the same.....')
+else:
+    length = len(reads[0]) 
+
+if args.reference is not None:
+    # A reference read has been provided.
+    referenceread = list(FastaReads(args.reference))[0]
+    mutationsdict = compareToRef(referenceread, reads, length)
+    # C{defaultdict} containing mutation counts.
+    totalmutations = sum(mutationsdict.values())
+    mutationsSorted = sorted(zip(mutationsdict.values(), mutationsdict.keys()), reverse=True)
```

Comment:
The `.keys()` here isn't needed. If you use a `dict` in an iterated context, you just get a list of its keys, just like calling `.keys()`.

Reply:
Done